### ISSUE TITLE
Any hints to better tree segmentation?

### ISSUE THREAD

**Issue by ShukhratSh** [SKrisanski/FSCT#29]
Hi Sean Krisanski,

Probably, it is the way the code works, but I just wanted to clarify to make sure I am not missing something. When you segment individual trees it looks like the code uses some buffer to attach points into an identified cylinder. This is causing an issue because bigger trees usually have smaller trees underneath, therefore parts of big trees just above the small trees are segmented as small trees. I tried to highlight what I meant in the screenshot below. 
It seems like the following part of the run.py is setting a threshold: 
`veg_sorting_range=1.5,  # Vegetation points can be, at most, this far away from a cylinder horizontally to be matched to a particular tree.`
Do you have any hints to avoid this?

![image](https://user-images.githubusercontent.com/11915606/198395115-bff36c94-c0ad-428b-999b-ef784dd873c4.png)

Thank you,
Shukhrat

**Comment by SKrisanski** [comment]
Hi Shukhrat,

This is a known limitation of the tool currently. There's some discussion about this in the FSCT paper if you want some more information about why it does this.

In short, I was often working with datasets that had major occlusions in the mid-story (such as combined above and below canopy UAS photogrammetry). I wasn't able to assume that the canopy was always connected to a stem, so there was a trade-off to be made. I could either have accurate tree heights for most major trees, at the cost of greatly overestimating the height of small trees under a tall canopy, or I could underestimate a lot of large trees' heights. I chose the first option in this case.
In the time I had to work on this problem, I hadn't worked out a nice solution that deals with all cases.
Going forward, I'll probably make some changes here and have 2 modes, but I've got pretty limited time to spend on FSCT currently. I am gradually refactoring the whole tool, so at some point in the next few months, the whole package will get a big upgrade.
All the best,
Sean
